src/hotspot/os/posix/os_posix.cpp

```diff
@@ -316,17 +316,21 @@ char* os::reserve_memory_aligned(size_t size, size_t alignment, int file_desc) {
 
   char* extra_base;
   if (file_desc != -1) {
-    // For file mapping, we do not call os:reserve_memory(extra_size, NULL, alignment, file_desc) because
-    // we need to deal with shrinking of the file space later when we release extra memory after alignment.
-    // We also cannot called os:reserve_memory() with file_desc set to -1 because on aix we might get SHM memory.
-    // So here to call a helper function while reserve memory for us. After we have a aligned base,
-    // we will replace anonymous mapping with file mapping.
+    // For file mapping, we do not call os:reserve_memory_with_fd since:
+    // - we later chop away parts of the mapping using os::release_memory and that could fail if the
+    //   original mmap call had been tied to a fd (TODO: really?)
+    // - there is no guarantee that os::reserve_memory even returns mmaped memory. In fact it never does
+    //   on Windows, obviously, and usually on AIX SystemV shm is used.
+    //
+    // Todo: this probably works more out of accident. Using reserve_mmapped_memory would require an munmap
+    //  to release, but later in this function os::release_memory is used which is not guaranteed to use mmap.
+    //  See JDK-8253851.
```

Comment:
Looks good apart from these two todo comments (i.e. the TODO:really? one and the last paragraph): they only replicate the information you added to JDK-8253851.
Source code isn't a good place to track issues, so I would prefer if these comments were removed since they seem superfluous. Your call.

Reply:
Thanks Thomas. 

I will remove the TODOs as you requested. As outside contributor I find those remarks very useful to make internal "xxx works on it" herd knowledge visible. But here I have no strong emotions.